tests/test_dad.py

```diff
@@ -0,0 +1,138 @@
+# Tests for the Data Analysis Designer (DAD)
+import pytest
+from django.urls import reverse
+
+from surveys import models
+
+
+@pytest.fixture
+def survey_submitted_setup(survey, survey_row, survey_component):
+    # Set up necessary fixtures for the surveys-submitted-detail view
+    pass
+
+
+def create_formset_data(initial, total, forms):
+    # Helper function to create POST data for a formset in the DAD.
+    # Arguments:
+    #   * initial (int): Initial number of forms on the page at load time
+    #   * total (int): Total number of forms on the page at POST time
+    #   * forms (iterable): Iterable of dictionaries, each one representing the
+    #                       field value for a form in the formset
+    output_form = {
+        # Initialize with management form data
+        'form-INITIAL_FORMS': initial,
+        'form-TOTAL_FORMS': total,
+    }
```

Comment:
Including both the `initial` and `total` arguments in this function might seem a little bit superfluous, particularly because in these tests we only ever set `total` to `len(forms)`. In that case, why not just set `total = initial + len(forms)`? In the end I decided to keep `total` as an explicit argument because I wanted to preserve the possibility of distinguishing between creating and editing charts, as in the following cases:

1. Test creating new charts -> `total == initial + len(forms)`
2. Test editing existing charts -> `total == initial`

Reply:
Makes sense! Thanks for the well documented function